src/resources/purchase/processor.ts

```diff
@@ -0,0 +1,161 @@
+import {
+  JsonApiParams,
+  KnexProcessor,
+  Operation,
+  Resource,
+  ResourceRelationship
+} from "@joelalejandro/jsonapi-ts";
+import * as MercadoPago from "mercadopago";
+import uuid = require("uuid");
+
+import Purchase from "./resource";
+
+import { MPPayerData } from "../../types";
+
+export default class PurchaseProcessor extends KnexProcessor<Purchase> {
+  public resourceClass = Purchase;
+
+  public async add(op: Operation) {
+    const purchase = { ...op.data };
+
+    // 1: Get all tickets to figure out quantity and total price.
+    const { totalPrice, tickets, quantitiesByType } = await this.getTickets(
+      purchase.relationships.ticket.data as ResourceRelationship[]
+    );
+
+    // 2: Create preference.
+    const preference = await this.postPreference({ tickets, quantitiesByType });
+
+    // 3: Create the purchase.
+    purchase.id = uuid.v4();
+    purchase.attributes = {
+      dateCreated: new Date().toJSON(),
+      status: "unpaid",
+      amountBilled: totalPrice,
+      externalId: preference.id
```

Comment:
To generate the payment URL I should use `https://www.mercadopago.com/mla/checkout/start?pref_id=<preference id>` right?

Reply:
Yes!